config/configgrpc/configgrpc.go

```diff
@@ -174,13 +174,13 @@ func (gcs *ClientConfig) isSchemeHTTPS() bool {
 // a non-blocking dial (the function won't wait for connections to be
 // established, and connecting happens in the background). To make it a blocking
 // dial, use grpc.WithBlock() dial option.
-func (gcs *ClientConfig) ToClientConn(ctx context.Context, host component.Host, settings component.TelemetrySettings, extraOpts ...grpc.DialOption) (*grpc.ClientConn, error) {
+func (gcs *ClientConfig) ToClientConn(_ context.Context, host component.Host, settings component.TelemetrySettings, extraOpts ...grpc.DialOption) (*grpc.ClientConn, error) {
 	opts, err := gcs.toDialOptions(host, settings)
 	if err != nil {
 		return nil, err
 	}
 	opts = append(opts, extraOpts...)
-	return grpc.DialContext(ctx, gcs.sanitizedEndpoint(), opts...)
+	return grpc.NewClient(gcs.sanitizedEndpoint(), opts...)
```

Comment:
Is this related to `go.opentelemetry.io/contrib/config`?

Reply:
Hmmm... in a previous version of this PR, an upgrade to a dep caused linting failures... hmmm... looks like in 1.63.2, the following deprecation notice was removed 🤔 

```
// Deprecated: use NewClient instead.  Will be supported throughout 1.x.
```

Reply:
See release notes, it was in .1: https://github.com/grpc/grpc-go/releases/tag/v1.63.1

Reply:
Split the change into a separate PR https://github.com/open-telemetry/opentelemetry-collector/pull/9965